addons/mail/models/res_partner.py

```diff
@@ -147,11 +147,16 @@ def _find_or_create_from_emails(self, emails, ban_emails=None,
         partners, tocreate_vals_list = self.env['res.partner'], []
         name_emails = [tools.parse_contact_from_email(email) for email in emails]
 
+        def in_ban_emails(email, ban_emails):
```

Comment:
Would define a small helper under ``_find_aliases`` as we are going to have to use it after each call to ``_find_aliases``